src/platform/uvc-option.h

```diff
@@ -101,13 +103,21 @@ class uvc_xu_option : public option
             uvc_range = ep->invoke_powered( [this]( platform::uvc_device & dev )
                                              { return dev.get_xu_range( _xu, _id, sizeof( T ) ); } );
 
-        if( uvc_range.min.size() < sizeof( int32_t ) )
+        if( uvc_range.min.size() < sizeof( T ) )
```

Comment:
Can you explain this if?

Reply:
yes - the range is a struct of 4 values: min, max, default step. In the code, these values were all pulled as int32_t. But sometimes, the value option's type is not int32_t - for example in our case, the value is int16_t.
So, obviously, the min, max, def step values should also have the same type.